CMakeLists.txt

```diff
@@ -193,11 +193,21 @@ scorec_export_library(core)
 #check for mallinfo2
 if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
   include(CheckCXXSymbolExists)
+  check_cxx_symbol_exists(mallinfo "malloc.h" PUMI_HAS_MALLINFO)
   check_cxx_symbol_exists(mallinfo2 "malloc.h" PUMI_HAS_MALLINFO2)
+  check_cxx_symbol_exists(mallctl "malloc_np.h" PUMI_HAS_MALLCTL)
+  if(PUMI_HAS_MALLINFO)
+    target_compile_definitions(core INTERFACE -DPUMI_HAS_MALLINFO)
+    target_compile_definitions(pcu PRIVATE -DPUMI_HAS_MALLINFO)
+  endif()
   if(PUMI_HAS_MALLINFO2)
     target_compile_definitions(core INTERFACE -DPUMI_HAS_MALLINFO2)
     target_compile_definitions(pcu PRIVATE -DPUMI_HAS_MALLINFO2)
   endif()
+  if(PUMI_HAS_MALLCTL)
+    target_compile_definitions(core INTERFACE -DPUMI_HAS_MALLCTL)
+    target_compile_definitions(pcu PRIVATE -DPUMI_HAS_MALLCTL)
+  endif()
 endif()
```

Comment:
@cwsmith Can we move this section into pcu/CMakeLists? Also, I don't think the core INTERFACE compile definition is necessary. The definition only applies to private code and test/describe.cc but I am going to fix that file to use PCU_GetMem().

Reply:
@bobpaw Sure.  That sounds good.

Reply:
Reading through `test/describe.cc`, I found a second function `get_chunk()` that returns the active memory + mmapped memory. I think that is analogous to `stats.mapped` on BSD[1][2]. But, since it requires more functionality than PCU.h provides, I think it makes the most sense to just make these compile definitions PUBLIC so that they propagate to the core target. Otherwise we would need to add another function to PCU.h (which might technically constitute a minor version bump).

Also, re-implementing the `get_heap()` function using `PCU_GetMem()` has hit a road bump. `PCU_GetMem()` does not indicate if the function is unavailable on the host OS. Right now, there is no `#else` clause, so it implicitly returns 0.0, which is potentially an actual memory amount. `get_heap()` returns -1.0 if the functionality is not supported.

 If you agree that making the definitions PUBLIC is the simplest thing, I'll just add the BSD (and Apple) code to `test/describe.cc`. It might be reasonable to notify users of core somehow that the `pumi_getMem()`/`PCU_GetMem()` function isn't supported, other than the compiler warning (in a separate PR).

https://github.com/SCOREC/core/blob/4cc2da9926adcca2a6b93ab0293040ac0e50049b/test/describe.cc#L21-L51
https://github.com/SCOREC/core/blob/4cc2da9926adcca2a6b93ab0293040ac0e50049b/test/describe.cc#L65-L84

1. https://www.man7.org/linux/man-pages/man3/mallinfo.3.html
2. https://man.freebsd.org/cgi/man.cgi?malloc(3)

Reply:
@bobpaw Where is the `get_heap()` function defined/called?  I didn't find it in `develop` (`git grep -i get_heap`).

Let's remove the additional memory queries/reporting from `test/describe.cpp` and just pass the result of  `PCU_GetMem()` into `print_stats(...)`.  I'm not aware of any users who rely on them.

I think a configure time warning is the best we can easily do to warn the users about that unimplemented `PCU_GetMem()`.  I agree, that should be in another PR.

Reply:
My bad, meant `get_peak()`. Otherwise sounds good.